improve get changed files

kcloutie opened the pull request:
# Changes

This change will add support for separating the files by the change type i.e. added, modified, deleted and renamed for a giving source control event. This provides the ability to target only added files or only renamed files. Note that this change still supports all changed files in case specifics types of changes are not needed.

In addition, this change will also allow the changed file information to be passed down using templating variables. The following templated variables have been added

- {{all_changed_files}}
- {{added_files}}
- {{deleted_files}}
- {{modified_files}}
- {{renamed_files}}

kcloutie committed:
improve get changed files

This change will add support for separating the files by the change
type i.e. added, modified, deleted and renamed for a giving
source control event. This provides the ability to target only added
files or only renamed files. Note that this change still supports all
changed files in case specifics types of changes are not needed.

In addition, this change will also allow the changed file information
to be passed down using templating variables. The following templated
variables have been added

- {{all_changed_files}}
- {{added_files}}
- {{deleted_files}}
- {{modified_files}}
- {{renamed_files}}

savitaashture commented:
/ok-to-test

piyush-garg commented:
lgtm

savitaashture commented:
@kcloutie PR looks good
Can we have a e2e for this change?

kcloutie commented:
> Can we have a e2e for this change?

I have not done anything in this area as of yet. Can I get some direction on what is required

kcloutie committed:
Merge branch 'main' into improve_changed_files3

kcloutie committed:
Merge branch 'main' into improve_changed_files3

chmouel commented:
So what about some refactoring on this to get:

{{ files. changed }}
{{ files. added }}
{{ files. deleted }}
{{ files. modified }}
{{ files. renamed }}

you can do this the same way we define body:

https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/cel/cel.go#L53

by defining a map[string]interface{} like we do on body  https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/cel/cel.go#L44

the advantage is that it become a CEL expression as well which we can do some funky stuff with

kcloutie commented:
> So what about some refactoring on this to get:
> 
> {{ files. changed }} {{ files. added }} {{ files. deleted }} {{ files. modified }} {{ files. renamed }}
> 
> you can do this the same way we define body:
> 
> https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/cel/cel.go#L53
> 
> by defining a map[string]interface{} like we do on body https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/cel/cel.go#L44
> 
> the advantage is that it become a CEL expression as well which we can do some funky stuff with

@chmouel  Is the idea for this proposed change to place all the changed files under `files` or a seperate section or to have it as a CEL expression? My change makes it available as a CEL expression already under a standard param. It would be under [pac](https://github.com/openshift-pipelines/pipelines-as-code/blob/d81b08ee860ab1d5820955510d6a1ee0851ed7f7/pkg/cel/cel.go#L59). i.e. `pac.all_changed_files` `pac.added_files` and so on. Below is a code example of e2e test I wrote using cel expressions

```
    pipelinesascode.tekton.dev/on-cel-expression: >-
      ( event == "push" &&  target_branch == "main" ) &&
      ( all_changed_files.matches('.tekton/pullrequest.yaml') ||  modified_files.matches('modified.txt') )
```

If you want to meet today to discuss, I can meet with you

chmouel commented:
> @chmouel Is the idea for this proposed change to place all the changed files under files or a seperate section or to have it as a CEL expression? My change makes it available as a CEL expression already under a standard param. It would be under [pac](https://github.com/openshift-pipelines/pipelines-as-code/blob/d81b08ee860ab1d5820955510d6a1ee0851ed7f7/pkg/cel/cel.go#L59). i.e. pac.all_changed_files pac.added_files and so on. Below is a code example of e2e test I wrote using cel expressions

having a CEL expression will be a side effect of doing it like this, I think this makes even easier than if you doing in another way.

We are not talking about CEL expression for filtering on annotations. I am talking about a CEL expression from the standard params.

ie folks would be able to do something like this in their tempalte:
{{ files.changed[0] })

to output the first file of the changed files

kcloutie commented:
Please see https://github.com/openshift-pipelines/pipelines-as-code/pull/1524
